### DISCUSSION TITLE
Module not found error

### DISCUSSION THREAD

**Discussion by u7068758** [cogent3/cogent3#1055] [Q&A]
Hi,
Although cogent3 has been successfully installed as shown below.
But when I try to perform dot plot in JupyterLab it gives the error: “No module named cogent3” (screenshot shown below)
Can you please guide me what’s the issue and how to solve?
![image](https://user-images.githubusercontent.com/107029444/172364198-2fd52ab4-0314-4701-bf60-ddadcf16da16.png)
![image](https://user-images.githubusercontent.com/107029444/172364316-ae93569c-c3c2-4dba-b92c-76ce8a425e12.png)

**Comment by GavinHuttley** [comment]
Hi @u7068758,

From the first screenshot showing terminal output from `pip install` it looks like you are running a `conda` environment. From this and the second screen grab (showing `JupyterLab`) I'm guessing the error is because `JupyterLab` is not connected to that `conda` environment. So I have a few questions, the answers to which will help us pin this down.

1. What version of Windows?
2. Are you running `conda` within the Windows Subsystem for Linux (WSL)?
3. How did you install `JupyterLab`?

**Reply by u7068758** [reply]
I'm using windows 10. Firstly, I installed Miniconda and then launched JupyterLab from Anaconda Navigator. Then I installed JupyterLab using the command "pip install JupyterLab" using command prompt App. The end result is as below:
![image](https://user-images.githubusercontent.com/107029444/172528651-efba8acb-c286-41d0-9b75-00b4a1944265.png)

**Comment by GavinHuttley** [comment]
I don't have a Windows machine, so I can't check this myself. Let me know if it works, or not!

First, quit `JupyterLab`. Then, in the terminal make sure the conda environment is active and install `ipykernel` using `pip`
```
$ pip install ipykernel
```
then create a kernel (this is a program that communicates between Python and `JupyterLab`)
```
$ python -m ipykernel install --sys-prefix --name python3 --display-name python3
```
When that's done, **in the terminal** start `JupyterLab`
```
$ jupyter lab
```
In a new Jupyer notebook try
```python
import cogent3
```

**Reply by u7068758** [reply]
Yup! that worked. Thanks.

**Reply by GavinHuttley** [reply]
To complete the experiment. Shutdown `JupyterLab` then launch it from the Navigator and see if the import still works. If it does, great, if it doesn't it means you will need to use the command line launcher.

**Reply by u7068758** [reply]
I tried this way. The error persists (screenshot shown below). What do you mean when you say to use the command line launcher?
![image](https://user-images.githubusercontent.com/107029444/172534322-cc843236-8cdb-416c-b3e4-6be1b36b1184.png)

**Reply by u7068758** [reply]
Sorry, I wrote cogent instead of cogent3 let me try again.

**Reply by u7068758** [reply]
Error persists.
![image](https://user-images.githubusercontent.com/107029444/172535359-b964662d-f8b9-40dd-9fb5-ea81b7d76275.png)

**Reply by GavinHuttley** [reply]
```
$ jupyter lab
```

**Reply by u7068758** [reply]
OK, got it
